app/controllers/queue_items_controller.rb

```diff
@@ -11,6 +11,18 @@ def create
     redirect_to my_queue_path
   end
 
+  def update_queues
+    update_reviews(params[:queue_ids_with_ratings]) if params[:queue_ids_with_ratings]
+    queue_ids_with_positions = params[:queue_ids_with_positions]
+    if queue_items_have_same_positions?(queue_ids_with_positions)
+      flash[:error] = 'Items cannot have same positions.'
+    else
+      flash[:error] = QueueItem.save_and_update_positions(queue_ids_with_positions)
```

Comment:
I have to say this solution is very complicated (in my opinion). I like the instructors solution better, also we call it position normalization.